scanpy/readwrite.py

```diff
@@ -277,15 +277,49 @@ def _read_v3_10x_h5(filename, *, start=None):
                 (data, dsets['indices'], dsets['indptr']),
                 shape=(N, M),
             )
+            obs_dict = dict(obs_names=dsets['barcodes'].astype(str))
+            var_dict = dict(
+                var_names=dsets['name'].astype(str),
+            )
+            if 'gene_id' not in dsets:
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['id'].astype(str),
+                        'feature_types': dsets['feature_type'].astype(str),
+                    }
+                )
+            else:
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['gene_id'].astype(str),
+                        'id': dsets['id'].astype(str),
+                        'feature_types': dsets['feature_type'].astype(str),
+                    }
+                )
+            if 'filtered_barcodes' in f['matrix']:
+                obs_dict.update(
+                    {'filtered_barcodes': dsets['filtered_barcodes'].astype(bool)}
+                )
+            if 'features' in f['matrix']:
+                feature_datasets_dict = {
+                    x: y.dtype.kind
```

Comment:
```suggestion
                var_dict.update({
                    x: dsets[x].astype(bool if y.dtype.kind == 'b' else str)                    
```
You can combine this two list comprehensions into one list comprehension.

Reply:
Had it this way in my first iteration and factored it out for readability. Fixed.